File: SanatoriumApp/SanatoriumApp.WebPortal/wwwroot/app/factory/auth/roleDefinitionProvider.js
'use strict'
angular.module('DataAccessApp').factory('roleDefinitionProvider', ['enumRoleIdentifier', '$filter', RoleDefinitionProvider]);

function RoleDefinitionProvider(enumRoleIdentifier, $filter) {

    var _roles = [
        {
            identifier: enumRoleIdentifier.admin,
            name: 'auth.role_admin',
        },
        {
            identifier: enumRoleIdentifier.medicalStaff,
            name: 'auth.role_medicalStaff',
        },
        {
            identifier: enumRoleIdentifier.utilityStaff,
            name: 'auth.role_utilityStaff',
        },
    ];

    return {
        getAll: function () {
            return _roles;
        },
        getByIdentifier: function (identifier) {
            return $filter('filter')(_roles, { identifier: identifier })[0];
        }
    };
}

File: SanatoriumApp/Core/Core.Domain/Auth/Roles/Definition/RoleDefinitionProvider.cs
using System.Collections.Generic;
using System.Linq;

namespace Core.Domain.Auth.Roles.Definition
{
    public class RoleDefinitionProvider : IRoleDefinitionProvider
    {
        private readonly ICollection<RoleDefinition> _definitions =
            new List<RoleDefinition>
            {
                new RoleDefinition
                {
                    Identifier = RoleIdentifier.Admin,
                },
                new RoleDefinition
                {
                    Identifier = RoleIdentifier.MedicalStaff,
                },
                new RoleDefinition
                {
                    Identifier = RoleIdentifier.UtilityStaff,
                },
            };

        public ICollection<RoleDefinition> GetAll()
        {
            return _definitions;
        }

        public RoleDefinition GetByIdentifier(RoleIdentifier identifier)
        {
            return _definitions.SingleOrDefault(d => d.Identifier == identifier);
        }
    }
}
